bluemira/base/components.py

```diff
@@ -190,6 +190,23 @@ def prune_child(self, name: str):
             # https://github.com/c0fec0de/anytree/issues/152
             found_component.parent = None
 
+    def _get_shapes(self):
+        """
+        Get all the shapes from a component
+        """
+        shapes = []
+        options = []
+        if self.is_leaf and isinstance(self, PhysicalComponent):
+            shapes.append(self.shape)
+            options.append(self.display_cad_options)
+        else:
+            for child in self.children:
+                shape, option = child._get_shapes()
+                shapes.extend(shape)
+                options.extend(option)
+
```

Comment:
1. The logic of getting a flat list of all the `PhysicalComponents` down the tree is a useful one for potentially other purposes. Can we put this in a separated function?

2. Is it possible to have a component without `PhysicalComponent` down the line ? If yes this function may error out or be stuck in a infinite loop?